modules/e2ee/E2EEncryption.js

```diff
@@ -244,11 +263,22 @@ export class E2EEncryption {
      * @param {*} newValue - The property's new value.
      * @private
      */
-    _onParticipantPropertyChanged(participant, name, oldValue, newValue) {
+    async _onParticipantPropertyChanged(participant, name, oldValue, newValue) {
         switch (name) {
         case 'e2ee.idKey':
             logger.debug(`Participant ${participant.getId()} updated their id key: ${newValue}`);
             break;
+        case 'e2ee.signatureKey':
+            logger.debug(`Participant ${participant.getId()} updated their signature key: ${newValue}`);
+            if (newValue) {
+                const parsed = JSON.parse(newValue);
+
+                const importedKey = await crypto.subtle.importKey('jwk', parsed, { name: 'ECDSA',
+                    namedCurve: parsed.crv }, true, parsed.key_ops);
+
+                this._e2eeCtx.setSignatureKey(participant.getId(), importedKey);
+            }
```

Comment:
shouldn't we set the key to false in case there is no new one?

Reply:
do we want to support removing your signature key?

Reply:
Great questions. I'd say the answer is no to both. Maybe we should at least log a warning when an empty key is published?

Reply:
for rejecting key updates we need worker support for sending back error message so lets do that late